src/status_im/contexts/wallet/common/utils.cljs

```diff
@@ -87,6 +75,40 @@
        (map #(calculate-balance-for-token %))
        (reduce +)))
 
+(defn calculate-balance-from-tokens
+  [{:keys [currency tokens]}]
+  (->> tokens
+       (map #(total-token-fiat-value currency %))
+       (reduce money/add)))
+
+(defn- add-balances-per-chain
+  [b1 b2]
+  {:raw-balance (money/add (:raw-balance b1) (:raw-balance b2))
+   :balance     (money/add (:balance b1) (:balance b2))
+   :chain-id    (:chain-id b2)})
+
+(defn- merge-token
+  [existing-token token]
+  (assoc token
+         :balances-per-chain
+         (merge-with add-balances-per-chain
+                     (:balances-per-chain existing-token)
+                     (:balances-per-chain token))))
+
+(defn aggregate-tokens-for-all-accounts
+  "Receives a seq of tokens per account (seq) and returns aggregated tokens"
+  [all-accounts-tokens]
+  (->> all-accounts-tokens
+       (reduce
+        (fn [result-map tokens-per-account]
+          (reduce
+           (fn [acc token]
+             (update acc (:symbol token) #(merge-token % token)))
+           result-map
+           tokens-per-account))
+        {})
```

Comment:
Let's make it a habit to have tests for all utility functions

Reply:
Yes. This utility function is used in the sub `:wallet/aggregated-tokens`.  So, I have added some subs tests. 👍

Reply:
@smohamedjavid About the sub, I think we could simplify the code a little, IMO understanding the logic of a `reduce` is not that hard, but two `reduce`s sharing the same acc, seems a little more difficult :sweat_smile: 

#### Suggestion

The code we have in the event can be seen as:

```clojure
(fn [accounts]
  (->> accounts
       (map :tokens)
       (utils/aggregate-tokens-for-all-accounts)))
```

and if we join the code in `utils/aggregate-tokens-for-all-accounts` to the previous code, the whole processing we are doing is:

```clojure
(fn [accounts]
  (->> accounts
       (map :tokens)
       (reduce (fn [result-map tokens-per-account] ;; code from utils/aggregate-tokens-for-all-accounts
                 (reduce (fn [acc token]
                           (update acc (:symbol token) #(merge-token % token)))
                         result-map
                         tokens-per-account))
               {})
       vals))
```

Here we can use `mapcat` instead of a double reduce:

```clojure
(->> accounts
     (mapcat :tokens)
     (reduce (fn [acc token]
               (update acc (:symbol token) #(merge-token % token)))
             {})
     vals)
```

also `update` receives a function and applies it to the previous value, so we can remove the `#(...)`:

```clojure
(->> accounts
     (mapcat :tokens)
     (reduce (fn [acc token]
               (update acc (:symbol token) merge-token token)) ;; Cleaner :)
             {})
     vals)
```

Reply:
Thanks for the suggestion @ulisesmac! I appreciate it.

My initial solution was using the `flatten`. I benchmarked it, and it was ~3ms slower than the double reduce. I tried using `mapcat`, but it's still the same.

```
"Elapsed time: 13.394000 msecs" <- Suggested Solution
"Elapsed time: 3.083625 msecs" <- Existing Solution
```
I'm concerned this number would grow if the user has plenty of tokens across the accounts. For the home page, It's best to keep the execution time low for a good and responsive UI. It's best to keep the double `reduce` as it's a bit faster. We can re-visit later if required.

I applied the rest of the suggestions in the functions.

Reply:
@smohamedjavid yes! those 13ms are too much!
Please consider adding a comment, so no one refactors those double reduce in the future :)